Quantum-Control-Applications/Superconducting/Single Fixed Transmon/readout_duration_opt.py

```diff
@@ -0,0 +1,221 @@
+"""
+optimal_weights.py: Optimal weights for the readout pulse
+"""
+from qm.qua import *
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from configuration import *
+import matplotlib.pyplot as plt
+import numpy as np
+from qm import SimulationConfig
+from qualang_tools.loops import from_array
+
+###########
+# Helpers #
+###########
+
+
+def divide_array_in_half(arr):
+    split_index = len(arr) // 2
+    arr1 = arr[:split_index]
+    arr2 = arr[split_index:]
+    return arr1, arr2
+
+
+def create_linear_array_of_same_length(arr):
+    length = len(arr)
+    return [i * 4 for i in range(length)]
+
+
+def create_complex_array(arr1, arr2):
+    return arr1 + 1j * arr2
+
+
+def plot_complex_array(arr):
+    plt.figure()
+    plt.plot(arr.real)
+    plt.plot(arr.imag)
+    plt.show()
+
+
+def subtract_complex_arrays(arr1, arr2):
+    return arr1 - arr2
+
+
+def normalize_complex_array(arr):
+    # Calculate the simple norm of the complex array
+    norm = np.sqrt(np.sum(np.abs(arr) ** 2))
+
+    # Normalize the complex array by dividing it by the norm
+    normalized_arr = arr / norm
+
+    # Rescale the normalized array so that the maximum value is 1
+    max_val = np.max(np.abs(normalized_arr))
+    rescaled_arr = normalized_arr / max_val
+
+    return rescaled_arr
+
+
+def plot_three_complex_arrays(arr1, arr2, arr3):
+    fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize=(15, 5))
+    ax1.plot(arr1.real)
+    ax1.plot(arr1.imag)
+    ax1.set_title("ground state")
+    ax2.plot(arr2.real)
+    ax2.plot(arr2.imag)
+    ax2.set_title("excited state")
+    ax3.plot(arr3.real)
+    ax3.plot(arr3.imag)
+    ax3.set_title("optimal weights")
+    plt.show()
+
+
+###################
+# The QUA program #
+###################
+
+number_of_divisions = 250  # number of chunks to divide the integration weights into // put something meaningful here
+division_length = int(readout_len / (4 * number_of_divisions))  # in clock cycles
```

Comment:
Wouldn't it make more sense to define the chunks by the chunk size instead of the number of chunks?
What happens if `readout_len / (4 * number_of_divisions)` is not an integer? for instance if readout length = 700ns and number_of_divisions=100?

Reply:
good point